### PR TITLE
eth: time out status message exchange after 5s

### PR DESCRIPTION
**Pull request by fjl**
This PR fixes an issue where "Gav's Node" would be stuck in the peer list even when a private network id was configured.

### CODE REVIEW

**Comment by robotally** [comment]
| Vote | Count | Reviewers |
| :---: | :---: | :---: |
| :+1: | 0 |  |
| :-1: | 0 |  |

_Updated: Thu Oct 22 20:25:23 UTC 2015_

**Comment by obscuren** [comment]
:+1:

File: eth/peer.go [hunk 1]
```diff
@@ -278,7 +280,26 @@ func (p *peer) Handshake(td *big.Int, head common.Hash, genesis common.Hash) err
 			GenesisBlock:    genesis,
 		})
 	}()
-	// In the mean time retrieve the remote status message
+	go func() {
+		errc <- p.readStatus(&status, genesis)
+	}()
+	timeout := time.NewTimer(handshakeTimeout)
+	defer timeout.Stop()
+	for i := 0; i < 2; i++ {
```
Comment (Gustav-Simonsson) [inline eth/peer.go]: Since `i` is not used, this can be done as a func...hmm...but then the `return` statements have to be forwarded.

**Comment by Gustav-Simonsson** [comment]
:+1:

**Pull request merged**